napytau/gui/components/logger.py

```diff
@@ -1,24 +1,121 @@
 import customtkinter
 
+from collections import deque
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
     from napytau.gui.app import App  # Import only for the type checking.
 
+# Define color constants for the logger
+LIGHT_MODE_INFO_COLOR = "black"
+LIGHT_MODE_ERROR_COLOR = "#ba1500"
+LIGHT_MODE_SUCCESS_COLOR = "#388f00"
+
+DARK_MODE_INFO_COLOR = "white"
+DARK_MODE_ERROR_COLOR = "#ff3a21"
+DARK_MODE_SUCCESS_COLOR = "#48ba00"
+
 
 class Logger(customtkinter.CTkFrame):
     def __init__(self, parent: "App") -> None:
         """
-        Initializes the logger frame.
+        The logger for the GUI.
         :param parent: Parent widget to host the logger.
         """
-        super().__init__(parent, height=100, corner_radius=10)
+        super().__init__(parent, height=10, corner_radius=10)
         self.parent = parent
+
+        self.grid(row=2, column=0, columnspan=1, padx=(10, 10), pady=(10, 10),
+                  sticky="ew")
         self.grid_propagate(False)
-        self.grid(row=2, column=0, columnspan=2, padx=10, pady=10, sticky="nsew")
 
-        # Label to display messages
-        self.output_label = customtkinter.CTkLabel(
-            self, text="Error messages etc. will be shown here", anchor="w"
+        # Create a scrollable frame for the log messages
+        self.scrollable_frame = customtkinter.CTkScrollableFrame(self, height=40)
+        self.scrollable_frame.pack(fill="both", expand=True, padx=10, pady=10)
+
+        # Set initial text colors
+        if customtkinter.get_appearance_mode() == "Light":
+            self.info_color = LIGHT_MODE_INFO_COLOR
+            self.error_color = LIGHT_MODE_ERROR_COLOR
+            self.success_color = LIGHT_MODE_SUCCESS_COLOR
+        else:
+            self.info_color = DARK_MODE_INFO_COLOR
+            self.error_color = DARK_MODE_ERROR_COLOR
+            self.success_color = DARK_MODE_SUCCESS_COLOR
+
+        # Store labels in a deque with a max length of 50
+        self.labels: deque = deque(maxlen=50)
+
+    def log_info(self, message: str) -> None:
+        """
+        Appends an informational message to the logger.
+        :param message: The message to log.
+        """
+        self._log_message("[INFO] " + message, color=self.info_color)
+
+    def log_error(self, message: str) -> None:
+        """
+        Appends an error message to the logger in red.
+        :param message: The error message to log.
+        """
+        self._log_message("[ERROR] " + message, color=self.error_color)
+
+    def log_success(self, message: str) -> None:
+        """
+        Appends a success message to the logger.
+        :param message: The message to log.
+        """
+        self._log_message("[SUCCESS] " + message, color=self.success_color)
+
+    def _log_message(self, message: str, color: str) -> None:
+        """
```

Comment:
This seems a bit complicated, why not have `log_message ` take the type of message as a parameter and define it as an enum (see: https://docs.python.org/3/library/enum.html for documentation on python enums)

You could then define a method on the enum to map to the string representation and call that in `log_message `

Reply:
Yeah, I like that idea.